OOP part 5/src/views/currency.view.ts

```diff
@@ -0,0 +1,203 @@
+
+    this.app.append(this.form);
+    this.app.before(h1tag);
+
+    document.querySelectorAll(".fromCurrency").forEach(box =>
+      box.addEventListener("change", (event) => this.changeToCurrencyValue(event))
+    )
+
+    document.querySelectorAll(".toCurrency").forEach(box =>
+      box.addEventListener("change", (event) => this.toCurrencyValue(event))
+    )
+
+    document.querySelector("#inputData").addEventListener('click', (event) => {
+      let fromFields = document.querySelectorAll(".fromCurrency");
+      fromFields.forEach((ele) => {
+        ele.setAttribute('type', 'text');
+      });
+
+      let toFields = document.querySelectorAll(".toCurrency");
+      toFields.forEach((ele) => {
+        ele.setAttribute('type', 'text');
+      });
+    });
+
+    document.querySelector("#rangeData").addEventListener('click', (event) => {
+      let fromFields = document.querySelectorAll(".fromCurrency");
+      fromFields.forEach((ele) => {
+        ele.setAttribute('type', 'range');
+      });
+
+      let toFields = document.querySelectorAll(".toCurrency");
+      toFields.forEach((ele) => {
+        ele.setAttribute('type', 'range');
+      });
+    });
+
+    document.querySelectorAll('input[type="range"][class="fromCurrency"]').forEach(box =>
+      box.addEventListener("change", (event) => console.log(event))
+    )
+
+  }
+
+  changeToCurrencyValue(event) {
+    let fromData = document.querySelectorAll(".fromCurrency");
+    let toData = document.querySelectorAll(".toCurrency");
+    let euroToRupeesData = fetch(`../rupeeConversion.json`, {}).then(response => response.json());
+    let euroToDollarsData = fetch(`../dollarConversion.json`, {}).then(response => response.json());
+    if (event.target.attributes[2].value == 'EuroToRupees') {
+      euroToRupeesData.then(res => { event.target.nextSibling.value = event.target.value * res['Rupees'] });
+    }
+    else {
+      euroToDollarsData.then(res => { event.target.nextSibling.value = event.target.value * res['Dollar'] });
+    }
+    if ((document.querySelector('input[name="mode"]:checked') as HTMLInputElement).value == 'ModeB') return;
+    fromData.forEach((ele) => {
+      ele.setAttribute('value', event.target.value);
+    });
+
+    euroToRupeesData.then(res => { toData[0].setAttribute('value', String(event.target.value * res['Rupees'])); });
+    euroToDollarsData.then(res => { toData[1].setAttribute('value', String(event.target.value * res['Dollar'])); });
+
+  }
+
+  toCurrencyValue(event) {
+    console.log(event);
+    let fromData = document.querySelectorAll(".fromCurrency");
+    let toData = document.querySelectorAll(".toCurrency");
+    let euroToRupeesData = fetch(`../rupeeConversion.json`, {}).then(response => response.json());
+    let euroToDollarsData = fetch(`../dollarConversion.json`, {}).then(response => response.json());
+    if (event.target.attributes[2].value == 'RupeesToEuro') {
```

Comment:
Make each function modular, like add this If condition in a separate function. Avoid doing many things in a single function